### PR TITLE
update default azure credential config

### LINKED ISSUE #9131: Identity AzureDefaultCredential Configuration.
Allow users to configure the credentials used in the Default Azure Credential chain.
Users should be able to exclude specific credentials from the chain.

### PR DESCRIPTION
**Pull request by g2vinay**
Fixes #9131

### CODE REVIEW

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java [hunk 1]
```diff
@@ -23,6 +34,57 @@ public DefaultAzureCredentialBuilder authorityHost(String authorityHost) {
...
+
+    /**
+     * Specifies whether the {@link EnvironmentCredential} will be excluded from the {@link DefaultAzureCredential}
+     * authentication flow. Setting to true disables reading authentication details from the process' environment
+     * variables.
+     *
+     * @param shouldExclude the status indicating to exclude the credential or not.
+     * @return An updated instance of this builder with the Environment credential exclusion set as specified.
+     */
+    public DefaultAzureCredentialBuilder excludeEnvironmentCredential(boolean shouldExclude) {
```
Comment (schaabs) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java]: Do you think that we need the boolean parameter? Should builder.excludeEnvironmentCredential() just set excludeEnvironmentCredential to true?

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java [hunk 2]
```diff
@@ -12,6 +17,12 @@
  * @see DefaultAzureCredential
  */
 public class DefaultAzureCredentialBuilder extends CredentialBuilderBase<DefaultAzureCredentialBuilder> {
+    private boolean excludeEnvironmentCredential;
+    private boolean excludeManagedIdentityCredential;
+    private boolean excludeSharedTokenCacheCredential;
+    private boolean excludeAzureCliCredential;
```
Comment (schaabs) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java]: In .NET the interactive browser credential is also part of the DefaultAzureCredential but it is excluded by default.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java (hunk 1 above)
Comment (g2vinay) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java]: Yeah, agreed and updated.
Just curious, in .NET I didn't see this pattern though.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java (hunk 2 above)
Comment (g2vinay) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java]: Yep, I'll be adding in separate PR and test it appropriately.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredential.java [hunk 3]
```diff
@@ -32,13 +32,9 @@
...
-     * @param identityClientOptions the options to configure the IdentityClient
+     * @param tokenCredentials the list of credentials to execute for authentication.
      */
-    DefaultAzureCredential(IdentityClientOptions identityClientOptions) {
-        super(new ArrayDeque<>(Arrays.asList(new EnvironmentCredential(identityClientOptions),
-            new ManagedIdentityCredential(null, identityClientOptions),
-            new SharedTokenCacheCredential(null, null, "04b07795-8ddb-461a-bbee-02f9e1bf7b46",
-                identityClientOptions),
-            new AzureCliCredential(identityClientOptions))));
+    DefaultAzureCredential(List<TokenCredential> tokenCredentials) {
```
Comment (jianghaolu) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredential.java:37]: Is there a reason to build a List in the builder and convert to ArrayDeque here?

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java [hunk 4]
```diff
@@ -50,6 +107,31 @@ public DefaultAzureCredentialBuilder executorService(ExecutorService executorSer
      * @return a {@link DefaultAzureCredential} with the current configurations.
      */
     public DefaultAzureCredential build() {
-        return new DefaultAzureCredential(identityClientOptions);
+        return new DefaultAzureCredential(getCredentialsChain());
+    }
+
+    private List<TokenCredential> getCredentialsChain() {
+        List<TokenCredential> output = new ArrayList<>(4);
```
Comment (jianghaolu) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java:114]: We can simply build a Deque here?

File: sdk/identity/azure-identity/src/test/java/com/azure/identity/DefaultAzureCredentialTest.java [hunk 5]
```diff
@@ -132,4 +132,84 @@ public void testNoCredentialWorks() throws Exception {
                                                    + "[\\$\\w]+\\$\\d*,\\s+AzureCliCredential[\\$\\w\\s\\.]+"))
             .verify();
     }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testExcludeCredentials() throws Exception {
+        DefaultAzureCredential credential = new DefaultAzureCredentialBuilder()
+                                                .excludeEnvironmentCredential()
```
Comment (jianghaolu) [inline sdk/identity/azure-identity/src/test/java/com/azure/identity/DefaultAzureCredentialTest.java:139]: too much indentation?

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java (hunk 1 above)
Comment (schaabs) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java]: in .NET we don't have builders or setters ;)

**Pull request merged**